genai-perf/genai_perf/main.py

```diff
@@ -165,9 +170,9 @@ def run():
             create_artifacts_dirs(args)
             tokenizer = get_tokenizer(args.tokenizer)
             generate_inputs(config_options)
-            args.func(args, extra_args)
+            telemetry_data_collector = args.func(args, extra_args)
```

Comment:
In the profile subcommand case, this line means that the call to the profile wrapper is returning the telemetry data collector.  That probably should not be coupled that way.